src/panels/lovelace/common/evaluate-filter.ts

```diff
@@ -1,12 +1,31 @@
 import { HassEntity } from "home-assistant-js-websocket";
+import { HomeAssistant } from "../../../types";
+
+export const evaluateFilter = (
+  hass: HomeAssistant,
+  current_entity_id: string,
+  filter: any
+): boolean => {
+  const stateObj: HassEntity = hass.states[current_entity_id];
+
+  if (!stateObj) {
+    return false;
+  }
 
-export const evaluateFilter = (stateObj: HassEntity, filter: any): boolean => {
   const operator = filter.operator || "==";
   let value = filter.value ?? filter;
   let state = filter.attribute
     ? stateObj.attributes[filter.attribute]
     : stateObj.state;
 
+  if (typeof value === "string" && value.includes(".")) {
```

Comment:
Should we maybe have used a new/different key other than `filter.value` for entity ids? Now I'm thinking that value string that has a `.` in it will no longer work since it will now be interpreted as an entity id, where maybe it was just supposed to be interpreted as a raw string. Like if it's filtering for state == "O.K." or "3.0" or something. 

Maybe instead of overloading filter.value with entity_ids, could have used filter.entity_id or filter.value_entity or something. Just something to think about that this might be breaking some existing usage.

Reply:
Oh yeah sure, reopening my first comment then.

As discussed [here](https://github.com/home-assistant/frontend/pull/19182#discussion_r1438345035): could `isValidEntityId` be enough ?

Will check

Reply:
isValidEntityId is probably not the right answer, as it's a very loose check, and it would still pass on things that may not be intended to be entities. 

IMO using a different key would be the correct solution.

Reply:
8c92954 should fix, as before changing the `value` value, it checks if the `value` is an `entity_id` part of `hass.states`.

Also using `isValidEntityId`: similar to `.includes(".")` but a regex to check word before and after the dot.

Reply:
I think core does similar things in automations, using the same field for entity id and a value, we should check what logic is used there and do the same.

Like:

```yaml
condition:
  condition: numeric_state
  entity_id: climate.living_room_thermostat
  attribute: temperature
  above: 17
  below: sensor.value
  ```

Reply:
I suspect that might only work because `above` is treated as a literal if it is a number type, or as an entity if it is a string type.

Not sure if there's a case where a value can either be a literal string or an entity string.

Reply:
True, we could fix this, and make it consistent by actually using the syntax we use in the automation conditions and the conditional card... So you would get:

```yaml
- type: entity-filter
  entities:
    - sensor.gas_station_auchan_saint_priest_e10
    - sensor.gas_station_carrefour_venissieux_e10
    - sensor.gas_station_relais_lyon_mermoz_e10
  state_filter:
    - condition: numeric_state
      above: 17
      below: sensor.value
  ```

Reply:
I don't think this quite address my concern as we don't have a matching syntax in automations for **non-numeric** state compare, where state can be either a string or an entity-id. Only numeric_state allows the field to be an entity_id. State condition does not allow that. 

But anyway it was just a minor suggestion, and I don't think it's a showstopper, so I guess whatever you settle on would be fine for most cases.

Reply:
Using consistent syntax (as automation or conditional card) is also making breaking changes for this card.

How could we handle that ?

Reply:
The old syntax should be removed from documentation but still functional in the card.

Reply:
Forgot to push but my last night commit about adding new conditions types, but.
It is kinda bigger than this one ... should we merge this PR before and make a separate PR for new conditions types or stack it to this one ?

Next:  `condition: numeric_state`, `condition: state` and `condition: operator`, which is the old syntax, backward compatible.

Reply:
@karwosts to get back to your (first) comment:

The automation's `NumericStateCondition` -> `above`/`below` are actually either a number or a string:

See how they are handled:
https://github.com/home-assistant/frontend/blob/dcb3accdb86f534277c16fd6d1c063f0a79c52dd/src/panels/config/automation/condition/types/ha-automation-condition-numeric_state.ts#L202-L208

You can try:
```yaml
condition: numeric_state
entity_id: sensor. gas_station_1
above: 1.1
below: sensor. gas_station_lowest_price
```

And `hass.states[entity_id]` is largely used in the codebase (256 results).

Reply:
Resolving this comment:
- uses the same codebase for `conditional` & `entity-filter` cards (`automation` can't be used as compute server side)
- properties `value`, `state`, `state_not`, `above`, `below` handle `entity_id` the same way.